Xamarin.Forms.Core/Button.cs

```diff
@@ -46,28 +46,34 @@ public class Button : View, IFontElement, ITextElement, IBorderElement, IButtonC
 		public static readonly BindableProperty CornerRadiusProperty = BindableProperty.Create("CornerRadius", typeof(int), typeof(Button), defaultValue: DefaultCornerRadius,
 			propertyChanged: CornerRadiusPropertyChanged);
 
-		public static readonly BindableProperty ImageProperty = BindableProperty.Create("Image", typeof(FileImageSource), typeof(Button), default(FileImageSource),
-			propertyChanging: (bindable, oldvalue, newvalue) => ((Button)bindable).OnSourcePropertyChanging((ImageSource)oldvalue, (ImageSource)newvalue),
-			propertyChanged: (bindable, oldvalue, newvalue) => ((Button)bindable).OnSourcePropertyChanged((ImageSource)oldvalue, (ImageSource)newvalue));
+		public static readonly BindableProperty ImageProperty = BindableProperty.Create(nameof(Image), typeof(FileImageSource), typeof(Button), default(FileImageSource),
+			propertyChanging: OnImageSourceChanging,
+			propertyChanged: OnImageSourceChanged);
+
 
 		public static readonly BindableProperty PaddingProperty = PaddingElement.PaddingProperty;
 
 		public Thickness Padding
 		{
-			get { return (Thickness)GetValue (PaddingElement.PaddingProperty); }
-			set { SetValue (PaddingElement.PaddingProperty, value); }
+			get { return (Thickness)GetValue(PaddingElement.PaddingProperty); }
+			set { SetValue(PaddingElement.PaddingProperty, value); }
 		}
 
-		Thickness IPaddingElement.PaddingDefaultValueCreator ()
+		Thickness IPaddingElement.PaddingDefaultValueCreator()
 		{
-			return default (Thickness);
+			return default(Thickness);
 		}
 
-		void IPaddingElement.OnPaddingPropertyChanged (Thickness oldValue, Thickness newValue)
+		void IPaddingElement.OnPaddingPropertyChanged(Thickness oldValue, Thickness newValue)
 		{
 			InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
 		}
 
+
+		internal static readonly BindablePropertyKey IsPressedPropertyKey = BindableProperty.CreateReadOnly(nameof(IsPressed), typeof(bool), typeof(Button), default(bool));
+		public static readonly BindableProperty IsPressedProperty = IsPressedPropertyKey.BindableProperty;
```

Comment:
This is new on Button no? Maybe add a UITest for it? and we need that on the notes